test/e2e/reboot.go

```diff
@@ -270,9 +272,15 @@ func waitForNodeToBe(c *client.Client, name string, wantReady bool, timeout time
 			// matches as desired.
 			if cond.Type == api.NodeReady && (cond.Status == api.ConditionTrue) == wantReady {
 				Logf("Successfully found node %s readiness to be %t", name, wantReady)
-				return true
+				flag = 0
+				return true, nil
 			}
+
 		}
+		return false, nil
+	}))
+	if flag == 0 {
```

Comment:
Similar comment to above. wait.Poll() returns an error if it failed, so you don't need this flag.